R/set-record-option.R

```diff
@@ -0,0 +1,110 @@
+
+#' Add or modify a record option
+#'
+#' Add a new record option or adjust the value/presence of an existing one.
+#'
+#' Note that only estimation, data, and table records are currently supported.
+#'
+#' Notes on `value`:
+#'  - We infer whether to use `option_value` or `option_flag` for *new* options based on the *class* of `value`:
+#'     - If you are trying to add an option with an assigned value (i.e. `MAXEVAL = 10`), `value` must be a character string or numeric value
+#'     - If you are trying to add a flag option (i.e. `NOABORT`), `value` should be logical (`T`/`F`).
+#'  - If `value` is set to `NULL`, the option will be removed (if found). `FALSE` is also acceptable for removing an existing option of class `nmrec_option_flag`.
+#'
+#' @param record An [nmrec_record] object.
+#' @param name Name of record to select. Any valid spelling of the option name
+#'   is allowed.
+#' @param value value to assign to option.
+#'
+#' @seealso `nmrec::option` selecting records of a given type.
+#'
+#'
+set_record_option <- function(record,
+                              name,
+                              value) {
+  stopifnot(inherits(record, "nmrec_record"))
+  stopifnot(length(name) == 1, is.character(name), nzchar(name))
+  stopifnot(record[["name"]] %in% c("estimation", "data", "table"))
+
+
+  name_map <- get(paste0(record[["name"]], "_option_names"))
+  name_resolved <- get0(tolower(name), name_map)
```

Comment:
This is missing the `sigma` and `omega` special cases (see `get_record_option`).  To get consistent handling of that (and avoid mismatches with future change) is part of the reason I [suggested](https://github.com/metrumresearchgroup/nmrec/pull/5#issuecomment-1587156188) you extract [get-record-option.R#L36-L44](https://github.com/metrumresearchgroup/nmrec/blob/f79c7c3c5e053a93448d002c4334d7343c3944bf/R/get-record-option.R#L36-L44) to a helper function.

Reply:
That was intentional. I thought we specifically didn't want this to be able to handle any `sigma` or `omega` cases?

Reply:
My [proposal](https://github.com/metrumresearchgroup/nmrec/pull/5#issuecomment-1587156188) is to handle all the "standard" options for any record type.  `sigma` and `omega` have a few.

(I know you originally meant for this to only support est/table/data records.  I should have used clearer wording above and said something like, "With the switch to support all record types, this is missing ...".)